cmake/configure/configure_2_trilinos.cmake

```diff
@@ -55,26 +55,18 @@ MACRO(FEATURE_TRILINOS_FIND_EXTERNAL var)
     ENDFOREACH()
 
     FOREACH(_optional_module
```

Comment:
@bangerth @niveshd I know how to fix the error. The problem is that we should check optional modules after do all the check for `trilinos` being ok. There are a lot below which make eventually disable `trilinos`. As a result we first might have `DEAL_II_TRILINOS_WITH_XYZ` set but then we actually disable Trilinos.

@niveshd please move the whole `FOREACH(_optional_module` part right after `CHECK_MPI_INTERFACE(TRILINOS ${var})` in this file:
```
IF(${var})
   FOREACH(_optional_module
     ....
   ENDFOREACH()
ENDIF()
```
this way we look for optional modules and enable them only if Trilinos has met all the rerquired dependencies above this line.

Reply:
p.s. `DEAL_II_FIND_FILE(ZOLTAN_CPP zoltan_cpp.h` shall not be needed as from the tester we see:
```
-- Found Zoltan
-- Found MueLu
-- Found ZOLTAN_CPP
-- Could not find a sufficient Trilinos installation: Trilinos has to be configured with the same MPI configuration as deal.II.
-- DEAL_II_WITH_TRILINOS has unmet external dependencies.
```